client/devdocs/design/playground-scope.js

```diff
@@ -0,0 +1,216 @@
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import FormStateSelector from 'components/forms/us-state-selector';
+import FormTelInput from 'components/forms/form-tel-input';
+import FormTextarea from 'components/forms/form-textarea';
+import FormTextInput from 'components/forms/form-text-input';
+import FormTextInputWithAction from 'components/forms/form-text-input-with-action';
+import FormTextInputWithAffixes from 'components/forms/form-text-input-with-affixes';
+import FormToggle from 'components/forms/form-toggle';
+import Gauge from 'components/gauge';
+import GlobalNotices from 'components/global-notices';
+import Gravatar from 'components/gravatar';
+import HeaderCake from 'components/header-cake';
+import HeaderButton from 'components/header-button';
+import ImagePreloader from 'components/image-preloader';
+import InfoPopover from 'components/info-popover';
+import InputChrono from 'components/input-chrono';
+import JetpackColophon from 'components/jetpack-colophon';
+import JetpackLogo from 'components/jetpack-logo';
+import LanguagePicker from 'components/language-picker';
+import ListEnd from 'components/list-end';
+import Main from 'components/main';
+import Notices from 'components/notice';
+import Pagination from 'components/pagination';
+import PaymentLogo from 'components/payment-logo';
+import PieChart from 'components/pie-chart';
+import Popovers from 'components/popover';
+import ProgressBar from 'components/progress-bar';
+import Ranges from 'components/forms/range';
+import Rating from 'components/rating';
+import Ribbon from 'components/ribbon';
+import ScreenReaderText from 'components/screen-reader-text';
+import Search from 'components/search';
+import SearchCard from 'components/search-card';
+import SectionHeader from 'components/section-header';
+import SectionNav from 'components/section-nav';
+import NavTabs from 'components/section-nav/tabs';
+import NavSegmented from 'components/section-nav/segmented';
+import NavItem from 'components/section-nav/item';
+import SegmentedControl from 'components/segmented-control';
+import SelectDropdown from 'components/select-dropdown';
+import ShareButton from 'components/share-button';
+import SiteTitleControl from 'components/site-title';
+import Spinner from 'components/spinner';
+import SpinnerButton from 'components/spinner-button';
+import SpinnerLine from 'components/spinner-line';
+import SplitButton from 'components/split-button';
+import Suggestions from 'components/suggestions';
+import TextDiff from 'components/text-diff';
+import TileGrid from 'components/tile-grid';
+import TimeSince from 'components/time-since';
+import Timezone from 'components/timezone';
+import TokenFields from 'components/token-field';
+import Tooltip from 'components/tooltip';
+import Version from 'components/version';
+import VerticalMenu from 'components/vertical-menu';
+import Wizard from 'components/wizard';
+
+export default {
```

Comment:
I recently learned we should avoid this syntax (`export default { ActionCard, Accordions }`) and do this instead:

```js
export ActionCard;
export Accordions;
// ...
```

```js
import * as playgroundScope from 'devdocs/design/playground-scope';
```

Context: p4TIVU-8Lf-p2 and https://github.com/Automattic/wp-calypso/pull/23820#issuecomment-382505982

Reply:
It's not likely to have a big impact here, but this is correct. We don't need to wrap everything in an object here. Also, we can be a bit more brief, there's a shorter way to export another module from your module:

```js
export ActionCard from 'components/action-card';
export Accordion from 'components/accordion';
// …
```

---

### More detail

_Often_, when we've done that historically, we wanted to export some names from our module, but exported and object as the default with those names as the properties.

It looks like that may not be the case here, and that the intent is to export an object, as is done here. This winds up wrapping the components up and passing the wrapper object to [`LiveProvider`](https://github.com/FormidableLabs/react-live#liveprovider-):

> Name | PropType | Description
> -- | -- | --
> scope | PropTypes.object | Accepts custom globals that the code can use

I wanted to make sure that we get and `Object` that behaves like an `Object` when we `import * from…` and I'm pretty sure we should be set to use that. Simple enough to try passing that to the `LiveProvider` and see what happens 🙂 

<details>
<summary>A truly painful amount of detail 🤓 </summary>
I've been trying to verify that when you `import * as someLocalName from '…'`, that it _is_ an object. With babel, it is. An [authoritative post also](http://2ality.com/2014/09/es6-modules-final.html#importing) says so:

> ```js
> // Importing the module as an object
> // (with one property per named export)
> import * as mylib from 'src/mylib';
> ```

[MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Import_an_entire_module's_contents) is a bit less clear, calling it a "namespace":

> […] the import * as name syntax imports all of them. Below are examples to clarify the syntax.
> 
> ### Import an entire module's contents
> This inserts myModule into the current scope, containing all the exports from the module in the file located in `/modules/my-module.js`.
> 
> ```js
> import * as myModule from '/modules/my-module.js';
> ```
> Here, accessing the exports means using the module name ("myModule" in this case) as a namespace. For example, if the module imported above includes an export doAllTheAmazingThings(), you would call it like this:
> 
> ```js
> myModule.doAllTheAmazingThings();
> ```

So… let's check [the ECMAScript® 2017
Language Specification](https://www.ecma-international.org/publications/files/ECMA-ST/Ecma-262.pdf):

> A module namespace object is an exotic object that exposes the bindings exported from an ECMAScript
Module (See 15.2.3). There is a one‑to‑one correspondence between the String‑keyed own properties of a
module namespace exotic object and the binding names exported by the Module. The exported bindings
include any bindings that are indirectly exported using export * export items. Each String‑valued own
property key is the StringValue of the corresponding exported binding name. These are the only String‑
keyed properties of a module namespace exotic object. Each such property has the attributes `{
[[Writable]]: true, [[Enumerable]]: true, [[Configurable]]: false }`. Module namespace objects are not
extensible.

An exotic object! I bet that's ok for our use case 👍 
</details>